### PR TITLE
LPS-198518: Avoid relate API Schemas that are not scoped as the API Endpoints

### PR DESCRIPTION
**Pull request by sergiojimcos**
## Motivation
The idea of this PR is to update `GROUP` property in API Application interface to `SITE`. You can see this property in `SCOPE_PICKLIST` for API Endpoint. With this, we can add a new validation in which if the Request Schema scope or the Response Schema scope (both of them from its `mainObjectDefintionERC`) are not scoped as its related API Endpoint, it will throw a Validation Exception.

**What are the changes?**
- Add new validation to carry out the motivation indicated above in `APIEndpointModelListener`.
- Update `headless-builder.json` for the creation of API Builder infrastructure in order to update `group` to `site`.
- Update APIApplication interface to update the `SITE` enum property.
- Updates impl, test and web modules. Please @jaimelr10 take a look at this if I make any change that is not correct.
- Update Poshi test --> feel free @yanan @Hijky and @magjed4289 to update this PR to change the Poshi tests.
- Add test case for that validation

**Which modules must deployed**
- Deploy `headless-builder-api`
- Deploy `headless-builder-impl`
- Deploy `headless-builder-web`

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPS-198518

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by magjed4289** [comment]
ci:reopen

**Comment by magjed4289** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx [hunk 1]
```diff
@@ -80,8 +80,8 @@ export default function BaseAPIEndpointFields({
 	);
 
 	const endpointPathHostTextPreview =
-		selectedScope?.value === 'group'
-			? `${window.location.origin}${basePath}${apiApplicationBaseURL}/scopes/${selectedScope.value}`
+		selectedScope?.value === 'site'
+			? `${window.location.origin}${basePath}${apiApplicationBaseURL}/scopes/{scopeKey}`
```
Comment (ccorreagg) [inline modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:84]: I think the update to {scopeKey} is done in purpose, but I am writing this comment just to be sure

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java [hunk 2]
```diff
@@ -533,6 +555,36 @@ public void test() throws Exception {
 				"headless-builder/endpoints", Http.Method.POST
 			).toString(),
 			JSONCompareMode.STRICT);
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"status", "BAD_REQUEST"
+			).put(
+				"title",
+				"There is an API schema that is not scoped like its related " +
```
Comment (ccorreagg) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:563]: Not sure if this description error is confusing :thinking: 

What about "The API Endpoint and the API Schema must be scoped by the same element" @4lejandrito any idea??

**Review by ccorreagg — COMMENTED** [review]
Just a couple of comments, but not sure if we should apply them or not

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java (hunk 2 above)
Comment (4lejandrito) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:563]: `The API Endpoint and the API Schema must be have the same scope`

Comment (ccorreagg) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:563]: Must have*

I like It! Thanks!

Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:563]: Me too :smile:

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx (hunk 1 above)
Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:84]: Yeah, this was a bug that was shown in the UI

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java (hunk 2 above)
Comment (4lejandrito) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:563]: :roll_eyes:

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
@sergiojimcos can you solve the conflicts and forward?

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by magjed4289** [comment]
ci:test:relevant

**Review by magjed4289 — APPROVED** [review]
3 Poshi tests needed an update, the rest of the failures is not related! ✅

**Comment by yanan** [comment]
Failures are not related(all updated poshi tests back to work), ok to forward :partying_face:

**Comment by yanan** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:forward:force

**Pull request closed without merging**